### PR TITLE
feat: new `@scalar/api-reference-editor` package

### PR DESCRIPTION
**Pull request by geoffgscott**
Adds an editor package to the repo to enable spec editing with internal or external state. 

Derived from: https://github.com/scalar/scalar/pull/1490

Also includes a fix to remove the npx call of `@scalar/cli` during build tooling which is dangerous.

### CODE REVIEW

**Comment by hanspagel** [comment]
I’ve extracted the new @scalar/galaxy build script changes to #2118, rebased the PR and added a changeset. ✌️

File: packages/api-reference-editor/src/components/EditorInput.vue [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+</script>
+
+<template>
+  <div
+    ref="codeMirrorRef"
+    class="editor-input">
+    <slot></slot>
+  </div>
+</template>
+<style>
```
Comment (marclave) [inline packages/api-reference-editor/src/components/EditorInput.vue:55]: lets scope this :)

File: packages/api-reference/src/types.ts [hunk 2]
```diff
@@ -250,7 +250,19 @@ export type ReferenceLayoutSlot =
   | 'sidebar-start'
   | 'sidebar-end'
 
+export type ReferenceLayoutSlots = {
+  [x in ReferenceLayoutSlot]: (props: ReferenceSlotProps) => any
```
Comment (marclave) [inline packages/api-reference/src/types.ts:254]: do we need to specify any here? maybe a comment would help : )

File: packages/api-reference/src/components/Layouts/Layouts.vue [hunk 3]
```diff
@@ -0,0 +1,39 @@
+<script setup lang="ts">
+/**
+ * This component allows dynamic selection of various layout configurations
+ */
+import type { ReferenceLayoutProps, ReferenceLayoutSlots } from '../../types'
+import ClassicLayout from './ClassicLayout.vue'
+import ModernLayout from './ModernLayout.vue'
+
+// eslint-disable-next-line vue/no-unused-properties
```
Comment (marclave) [inline packages/api-reference/src/components/Layouts/Layouts.vue:9]: should we just pick the props we use from this?

**Review by marclave — CHANGES_REQUESTED** [review]
i get an error for `[vite-node] [ERR_LOAD_URL] ./nuxt-theme.css` on nuxt 

also we should add some documentation to root README (even if just a link to the package readme) and remove our `isEditable` prop

File: packages/api-reference/src/components/Layouts/Layouts.vue (hunk 3 above)
Comment (geoffgscott) [reply packages/api-reference/src/components/Layouts/Layouts.vue:9]: No we are passing props through.

File: packages/api-reference/src/types.ts (hunk 2 above)
Comment (geoffgscott) [reply packages/api-reference/src/types.ts:254]: I'm not sure this is previous code that was moved.

File: packages/api-reference-editor/src/components/EditorInput.vue (hunk 1 above)
Comment (geoffgscott) [reply packages/api-reference-editor/src/components/EditorInput.vue]: Done.

**Review by marclave — APPROVED** [review]

**Pull request merged**